shell/components/SortableTable/index.vue

```diff
@@ -900,6 +900,7 @@ export default {
                     <td
                       :key="col.col.name"
                       :data-title="col.col.label"
+                      :data-cy="col.col.name"
```

Comment:
Aren't the e2e tests and unit tests for the rancher ui written via cypress?  If so, then I guess data-cy would be an appropriate name, but to keep it consistent with other components, I guess I should change it to data-testid

Reply:
Unit tests are way too slow with Cypress and lack of some tools, so we use Jest as pretty much the standard for the scope.

Reply:
Could you please still provide a meaningful name as suggested and for the mentioned reasons, e.g.: 

```
:data-testid="`sortable-cell-${ i }-${ j }`"
```

You will have to update the opened template loop for the columns as well, like this

```html
<template v-for="(col, j) in row.columns">
```

For the rows instead, the index is already defined as `i` 

```html
<template v-for="(row, i) in group.rows">
```

Reply:
However, if we change it like this. When two columns in the list are swapped, then we need to update the cypress code synchronously.

Reply:
We need to pay attention to the values of `i` and `j` when writing the cypress code, which I think is not necessary。

Reply:
Ofc you should be careful with that, because the test have to be deterministic.

If you want to rely on names, then you directly use the content value. In the Cypress best practices documentation they mention this very case: https://docs.cypress.io/guides/references/best-practices#Text-Content

Reply:
got it， thanks

Reply:
I read this link for best practices, but I think the purpose of adding data-testid to the table is to quickly locate the column you want to find, and then use cy.contains() to get the content of the element after using data-testid to find td in tr, and we don't need to care if this column is in first or second place, because it does not affect the jump to determine whether the resource was created successfully.

Reply:
In the guidelines says clearly when to use the content and when the ID.

Could you please explain what is the problem of not using the name vs using it? I am not really understanding what is your concern about this.